src/Integration.Vsix/CFamily/FileConfig.cs

```diff
@@ -30,19 +30,36 @@ partial class CFamilyHelper
     {
         internal class FileConfig
         {
-            public static FileConfig TryGet(ProjectItem dteProjectItem, string absoluteFilePath)
+            public static FileConfig TryGet(ILogger logger, ProjectItem dteProjectItem, string absoluteFilePath)
             {
                 var vcProject = dteProjectItem.ContainingProject.Object as VCProject;
                 var file = dteProjectItem.Object as VCFile;
                 if (vcProject == null || file == null)
                 {
                     return null;
                 }
-
                 var vcConfig = vcProject.ActiveConfiguration;
+                var projectKind = vcConfig.ConfigurationType;
+                // Unknown projects represent all the unspported project type like makefile projects
+                if (projectKind == ConfigurationTypes.typeUnknown)
+                {
+                    logger.WriteLine(@"Project's ""Configuration type"" is not supported.");
+                    return null;
+                }
+                // "ClCompile" for source files and "ClCompile" for header files
+                if (file.ItemType != "ClCompile")
+                {
+                    logger.WriteLine($"File's \"Item type\" is not supported. File: '{absoluteFilePath}'");
+                    return null;
+                }
                 var platformName = ((VCPlatform)vcConfig.Platform).Name; // "Win32" or "x64"
                 var vcFileConfig = file.GetFileConfigurationForProjectConfiguration(vcConfig);
-
+                // We don't support custom build tools. VCCLCompilerTool is needed for all the necessary compilation options to be present
+                if (!(vcFileConfig.Tool is VCCLCompilerTool))
+                {
+                    logger.WriteLine($"Custom built files are not supported. File: '{absoluteFilePath}'");
```

Comment:
I'm not a c++ dev so I don't know if this message is clear enough. Perhaps something like "File {0} is built using custom build tools. SonarLint supports only VCCLCompilerTool". 🤔

Reply:
VCCLCompilerTool is an internal name that won't resonate with C++ dev.  I prefer when I can to keep the variable size name at the end to avoid bad messages when the absolute path is big. 
maybe:
Custom build tools aren't supported. Custom-built file: '{absoluteFilePath}'
What do you think?

Reply:
Yea that can work 👍 but whatever you think will be most understandable to c++ devs.